docker/Dockerfile

```diff
@@ -41,47 +41,78 @@ ARG MOTD='printf "\n\
  This Docker container comprises of a WSO2 product, running with its latest GA release \n\
  which is under the Apache License, Version 2.0. \n\
  Read more about Apache License, Version 2.0 here @ http://www.apache.org/licenses/LICENSE-2.0.\n"'
-ENV ENV=${USER_HOME}"/.bashrc"
-
-# create the non-root user and group and set MOTD login message
-RUN \
-    groupadd -r -g ${USER_GROUP_ID} ${USER_GROUP} \
-    && useradd -m -r -u ${USER_ID} -d ${USER_HOME} -g ${USER_GROUP} ${USER} \
-    && echo ${MOTD} > "${ENV}"
-
-# To avoid warning message due to APIIUtils
-ENV DEBCONF_NOWARNINGS yes
-
+ENV ENV=${USER_HOME}"/.ashrc"
+
+# GLIB based alpine image (alpine-glib) source is generously borrowed
+# From https://github.com/Docker-Hub-frolvlad/docker-alpine-glibc
+# Here we install GNU libc (aka glibc) and set C.UTF-8 locale as default.
+RUN ALPINE_GLIBC_BASE_URL="https://github.com/sgerrand/alpine-pkg-glibc/releases/download" && \
+    ALPINE_GLIBC_PACKAGE_VERSION="2.30-r0" && \
+    ALPINE_GLIBC_BASE_PACKAGE_FILENAME="glibc-$ALPINE_GLIBC_PACKAGE_VERSION.apk" && \
+    ALPINE_GLIBC_BIN_PACKAGE_FILENAME="glibc-bin-$ALPINE_GLIBC_PACKAGE_VERSION.apk" && \
+    ALPINE_GLIBC_I18N_PACKAGE_FILENAME="glibc-i18n-$ALPINE_GLIBC_PACKAGE_VERSION.apk" && \
+    apk add --no-cache --virtual=.build-dependencies wget ca-certificates && \
+    echo \
+        "-----BEGIN PUBLIC KEY-----\
+        MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEApZ2u1KJKUu/fW4A25y9m\
+        y70AGEa/J3Wi5ibNVGNn1gT1r0VfgeWd0pUybS4UmcHdiNzxJPgoWQhV2SSW1JYu\
+        tOqKZF5QSN6X937PTUpNBjUvLtTQ1ve1fp39uf/lEXPpFpOPL88LKnDBgbh7wkCp\
+        m2KzLVGChf83MS0ShL6G9EQIAUxLm99VpgRjwqTQ/KfzGtpke1wqws4au0Ab4qPY\
+        KXvMLSPLUp7cfulWvhmZSegr5AdhNw5KNizPqCJT8ZrGvgHypXyiFvvAH5YRtSsc\
+        Zvo9GI2e2MaZyo9/lvb+LbLEJZKEQckqRj4P26gmASrZEPStwc+yqy1ShHLA0j6m\
+        1QIDAQAB\
+        -----END PUBLIC KEY-----" | sed 's/   */\n/g' > "/etc/apk/keys/sgerrand.rsa.pub" && \
```

Comment:
Is this key has a expiry time

Reply:
This is the public key for the glibc for alpine software repository. This might expire but that doesn't effect the released docker images. This is just a build time dependency for fetching the required glibc packages. After that there is no use of this cert. So if it expires we can update it to latest when we are building the latest image.